packages/client/src/AddSite.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useState } from 'react';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogContentText from '@material-ui/core/DialogContentText';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import TextField from '@material-ui/core/TextField';
+import Button from "@material-ui/core/Button";
+
+
+import { gql, useMutation, useQuery} from '@apollo/client';
+
+const CREATE_SITE = gql`
+  mutation CreateSite($site: SiteInput!) {
+    createSite(site: $site) {
+      siteURL
+      id
+    }
+  }
+`;
+
+export default function AddSite(props: any) {
+
+
+  // Todo: Finish Validator
+  const validURL = (str) => {
+    var pattern = new RegExp('^(https?:\\/\\/)?'+ // protocol
+      '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|'+ // domain name
+      '((\\d{1,3}\\.){3}\\d{1,3}))'+ // OR ip (v4) address
+      '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*'+ // port and path
+      '(\\?[;&a-z\\d%_.~+=-]*)?'+ // query string
+      '(\\#[-a-z\\d_]*)?$','i'); // fragment locator
+    return true || !!pattern.test(str);
```

Comment:
I believe the issue is the validURL doesn't like strings including http://localhost:9999 , we should update the validator to accept that string also.  This would be a great function to unit test.